Scala/Scala.sublime-syntax

```diff
@@ -813,6 +857,7 @@ contexts:
       scope: support.type.scala
     - include: base-type-expression
   single-type-expression:
+    - include: annotation
```

Comment:
Thanks for the review.
I just tried with Scala 2.10 and 2.11 and `(e: Int @unchecked)` compiles but `(e: @unchecked Int)` doesn't ( "error: ')' expected but identifier found." )
The `annotation` context doesn't popup after the annotation and we still are in the `single-type-expression` so it works with  `(e: Int @unchecked)`.
With `(e: @unchecked Foo)` the `Foo` appears as a `support.constant` when the user would expect a type, which hints the compile error.